### ISSUE TITLE
"Cannot find module eslint-config-*" when using yarn plug n play

### ISSUE THREAD

**Issue by camjackson** [AtomLinter/linter-eslint#1211]
## Issue Type
Bug

## Issue Description
1. I created a new project using `npx create-react-app --use-pnp`, and then added some eslint configs/plugins. 
2. I can run `yarn eslint src` and successfully lint my project
3. But linter-eslint always says: `Cannot find module 'eslint-config-react-app'

Suspect that this is because yarn pnp does weird stuff with module resolution. If I try to run eslint from the CLI via npm (instead of yarn), it also fails to load modules etc.

So, I don't know how linter-eslint attempts to load eslint and its configs/plugins, but if it's assuming that the packages will all be there in node_modules, it will fail. Any attempt to load modules needs to be done from within a yarn process, so that yarn can do its thing with module resolution.

`Linter Eslint: Debug`:
```
Atom version: 1.33.1
linter-eslint version: 8.4.1
ESLint version: 4.19.1
Hours since last Atom restart: 0.2
Platform: win32
Using bundled fallback ESLint from: C:\Users\Cam\.atom\packages\linter-eslint\node_modules\eslint
Current file's scopes: [
  "source.json",
  "meta.structure.dictionary.json",
  "meta.structure.dictionary.value.json",
  "punctuation.separator.dictionary.pair.json"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDirs": [],
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```

**Comment by anthonator** [comment]
We recently upgraded to Yarn v2 and we've started running into this as well.

Just to note. `pnp` is now default for Yarn v2. I imagine this will become more of a problem for people going forward.

**Comment by IanVS** [comment]
As far as I can tell, this error isn't because of pnp, but rather because you don't have your own version of eslint installed in your project, which causes linter-eslint to fall back to the version it bundles internally.  However, you can't use that version if you're also using custom configs and plugins.  Please try installing `eslint` in your dev dependencies and let us know if you continue to have problems.

**Comment by anthonator** [comment]
Wo do have `eslint` installed as a dev dependency.

**Comment by IanVS** [comment]
I was responding to the original poster.  Can you please post the output from the `Linter Eslint: Debug` command, @anthonator?

**Comment by IanVS** [comment]
Ah, yes it looks like we are looking for eslint in `node_modules` ([debug command](https://github.com/AtomLinter/linter-eslint/blob/cf8444107576f3bc80a131d17a31b241c56742e6/src/worker.js#L78), [actual resolution](https://github.com/AtomLinter/linter-eslint/blob/6c1ecd89c84a24b22450c2c8d324a16999641d45/src/worker-helpers.js#L58)).  Is this something you'd be willing to work on a PR for?  I'm not using yarn pnp (or even Atom for that matter), so it would be great for someone to work on this and I'm sure I and @Arcanemagus can provide some guidance along the way.

**Comment by anthonator** [comment]
I'm not sure this is something I currently have time for.

Is it possible to specify the command that runs ESlint? If not would it be possible to add something like that? Not sure if that would be easier and provide a "good enough" solution for the time being.

**Comment by zypA13510** [comment]
I have looked into this yesterday. And to my understanding, `@yarnpkg/pnpify` mentioned in [Editor SDKs](https://yarnpkg.com/advanced/editor-sdks) needs to be set up by the user, and then linter-eslint needs to add the path to its recognized eslint locations. However, I think we should wait for yarnpkg/berry#1231 to be resolved first.

Btw, I have added test & fixture at my fork in case anyone is interested: 1ef260233b728e72ec924924c5b51139ad4cd600
